yjit/src/core.rs

```diff
@@ -1875,11 +1879,16 @@ fn get_branch_target(
 
         // Add an incoming branch into this block
         block.push_incoming(branchref.clone());
-        let mut branch = branchref.borrow_mut();
-        branch.blocks[target_idx.as_usize()] = Some(blockref.clone());
 
-        // Return a pointer to the compiled code for the block
-        return block.start_addr;
+        // Fill out the target with this block
+        branch.targets[target_idx.as_usize()] = Some(Box::new(BranchTarget {
```

Comment:
I assume `Box::new` allocates this on (a different slot of) the heap. When is it freed? Is it freed when the `branch` is deallocated?

Reply:
Right. It's analogous to how when a Block is dropped the incoming and outgoing vecs are dropped with it.